### PR TITLE
Map upgrade

### PR DESCRIPTION
**Pull request by jaka**
Now the „big“ map fetches the same data as the map from nodewatcher.

### CODE REVIEW

**Comment by mitar** [comment]
How are you testing this? Was this the only node which was using nodewatcher?

File: mainpage/wlansi/static/wlansi/js/map.js [hunk 1]
```diff
@@ -49,4 +72,6 @@ $(document).ready(function () {
         }, true));
     }
     google.maps.event.addListener(map, 'bounds_changed', updateStatus);
+
```
Comment (mitar) [inline mainpage/wlansi/static/wlansi/js/map.js]: Why extra lines?

File: mainpage/wlansi/static/wlansi/js/map.js [hunk 2]
```diff
@@ -23,24 +26,44 @@ $(document).ready(function () {
...
             'shape': shape,
-            'title': node.name
-        });
-        google.maps.event.addListener(marker, 'click', function () {
-            document.location = node.url;
-        });
-    });
+            'title': node.n
+          });
+          (function(uuid){google.maps.event.addListener(marker, 'click', function () {
```
Comment (mitar) [inline mainpage/wlansi/static/wlansi/js/map.js]: Keep coding style. Spaces around:

```
function (uuid) {google.maps.event.addListener(marker, 'click', function () {
```

Also, no need to make one-liners. Target on code readability.

In this particular case you could use `$.each` to iterate over `nodes` which would then solve the problem of variable aliasing in JavaScript.

File: mainpage/wlansi/static/wlansi/js/map.js [hunk 3]
```diff
@@ -23,24 +26,44 @@ $(document).ready(function () {
...
+    $.get(nodewatcher_url + "/api/v1/stream/?format=json&tags__module=topology&limit=1", function (data) {
+       var streamId = data.objects[0].id;
+       var latestTimestamp = Math.round(Date.parse(data.objects[0].latest_datapoint) / 1000);
+       $.getJSON(nodewatcher_url + "/api/v1/stream/" + streamId + "/?format=json&reverse=true&limit=1&start=" + latestTimestamp, function(x) {
+        nodes = x.datapoints[0].v.v;
+        displayNodes();
+      });
+    });
+
+    function displayNodes() {
```
Comment (mitar) [inline mainpage/wlansi/static/wlansi/js/map.js]: Define functions you are calling before you are calling them. :-)

**Comment by mitar** [comment]
Can you filter only up nodes?

**Comment by mitar** [comment]
This displays only up nodes? This is good.

**Comment by jaka** [comment]
This displays the same nodes as nodewatcher that have coordinates. Currently it also displays _backbone_ nodes. One condition if cause more and it will show only _wireless_ nodes.

**Comment by mitar** [comment]
No, I would like not wireless nodes, but "up" nodes. Not sure if this is even possible currently to make such a query. @kostko?

**Comment by mitar** [comment]
BTW, when you push to the pull request, I do not get a notification. So if you would like me to check new things, in general, write a comment requesting that.

File: mainpage/wlansi/static/wlansi/js/map.js [hunk 4]
```diff
@@ -23,30 +26,47 @@ $(document).ready(function () {
         'mapTypeId': google.maps.MapTypeId.ROADMAP,
     };
     var map = new google.maps.Map(document.getElementById('map_canvas'), map_options);
-    $.each(nodes, function (i, node) {
-        var marker = new google.maps.Marker({
-            'position': new google.maps.LatLng(node.lat, node.long),
+
+    function updateStatus() {
+      var bounds = map.getBounds();
```
Comment (mitar) [inline mainpage/wlansi/static/wlansi/js/map.js]: Do we now use 4 spaces or 2 spaces for indent? ;-)

File: mainpage/wlansi/static/wlansi/js/map.js [hunk 5]
```diff
@@ -23,30 +26,47 @@ $(document).ready(function () {
...
+          });
+          google.maps.event.addListener(marker, 'click', function () {
+            document.location = nodewatcher_url + "/node/" + node.i + "/";
+          });
+        }
+      });
+      updateStatus();
     }
+
+    $.get(nodewatcher_url + "/api/v1/stream/?format=json&tags__module=topology&limit=1", function (data) {
```
Comment (mitar) [inline mainpage/wlansi/static/wlansi/js/map.js]: Use `getJSON` here as well.

Comment (mitar) [inline mainpage/wlansi/static/wlansi/js/map.js]: Also, I prefer passing query string as data argument (second argument). From [documentation](https://api.jquery.com/jquery.getjson/#jQuery-getJSON-url-data-success):

> Data that is sent to the server is appended to the URL as a query string. If the value of the data parameter is a plain object, it is converted to a string and url-encoded before it is appended to the URL.

File: mainpage/wlansi/map/templates/map/map.html [hunk 6]
```diff
@@ -4,11 +4,6 @@
...
 {% addtoblock "js" %}<script type="text/javascript">
     /* <![CDATA[ */
-    var nodes = [
-        {% for node in nodes %}
-            {'lat': {{ node.geo_lat|unlocalize }}, 'long': {{ node.geo_long|unlocalize }}, 'name': '{{ node.name|escapejs }}', 'url': '{{ node.get_full_url|escapejs }}'}{% if not forloop.last %},{% endif %}
-        {% endfor %}
-    ];
     var MAP_LATITUDE = '{{ MAP_LATITUDE|escapejs }}';
     var MAP_LONGITUDE = '{{ MAP_LONGITUDE|escapejs }}';
     var MAP_ZOOM = {{ MAP_ZOOM }};
```
Comment (mitar) [inline mainpage/wlansi/map/templates/map/map.html]: You could create URLs to icons here as well: https://stackoverflow.com/questions/27932983/django-static-path-in-javascript-file

File: mainpage/wlansi/static/wlansi/js/map.js (hunk 5 above)
Comment (jaka) [inline mainpage/wlansi/static/wlansi/js/map.js]: This code is actually copy-paste from nodewatcher map script.

Comment (mitar) [inline mainpage/wlansi/static/wlansi/js/map.js]: All our code is continuously evolving. :-) We all learn all the time new stuff. And then when we notice old code around, we can fix it.

Comment (mitar) [inline mainpage/wlansi/static/wlansi/js/map.js]: Feel empowered to fix and improve on any code you see. No matter who made it.

**Comment by mitar** [comment]
I think this is getting to look very good. Also our map on nodewatcher would need some love. See [here how our old map looked](http://old.wlan-si.net/) and people are missing features. You could look into this as well. (Oh, I see that it does not work anymore. Hm, why.)

(Two main features which are currently missing is that you cannot click on node and get some bubble with basic information. And that in URL you do not have anymore current state encoded so you cannot share what you see on the map with somebody else. We encoded zoom and position and filters into anchor.)

**Comment by mitar** [comment]
One more thing. When you are making pull request. In your fork you should always keep upstream branches intact. And create your own branches into which you make changes. So do not work directly in `master` branch of your fork, but create a so-called feature branch and work there. The reason is that if for some reason we do not merge your branch as it is, then your `master` branch gets out of sync with this `master` branch and from then on you cannot really make any reasonable pull requests, because you will always have those unmerged commits in.

**Comment by mitar** [comment]
I merged it. I made some additional commits with smaller style changes.

I made this ticket with more ideas for improvements to this map: https://grow.wlan-si.net/ticket/1343

Anyway, amazing job! Thanks!

**Comment by mitar** [comment]
BTW, you can ignore those TODOs about hard-coded URLs. We do not really have a good solution for that at the moment.

**Comment by mitar** [comment]
I pushed it to the production. Great job!

**Comment by jaka** [comment]
I see it working ... maybe it didn't because nw had problems.

**Comment by mitar** [comment]
Ah, I have a plugin which blocks 3rd party scripts. Cool!

**Pull request merged**